python/openai/openai/tests/conftest.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from pathlib import Path
+
+import pytest
+from fastapi.testclient import TestClient
+from tests.utils import OpenAIServer, setup_fastapi_app
+
+### TEST ENVIRONMENT SETUP ###
+TEST_BACKEND = ""
+TEST_MODEL = ""
+TEST_PROMPT = "What is machine learning?"
+TEST_MESSAGES = [{"role": "user", "content": TEST_PROMPT}]
+TEST_TOKENIZER = "meta-llama/Meta-Llama-3.1-8B-Instruct"
+
+# Infer the test environment for simplicity in local dev/testing.
+try:
+    import vllm as _
+
+    TEST_BACKEND = "vllm"
+    TEST_MODEL = "llama-3.1-8b-instruct"
+except ImportError:
+    print("No vllm installation found.")
+
+try:
+    import tensorrt_llm as _
+
+    TEST_BACKEND = "tensorrtllm"
+    TEST_MODEL = "tensorrt_llm_bls"
+except ImportError:
+    print("No tensorrt_llm installation found.")
+
+if not TEST_BACKEND or not TEST_MODEL:
+    raise Exception("Unknown test environment")
+###
+
+
+# NOTE: OpenAI client requires actual server running, and won't work
+# with the FastAPI TestClient. Run the server at module scope to run
+# only once for all the tests below.
+@pytest.fixture(scope="module")
+def server():
+    model_repository = Path(__file__).parent / f"{TEST_BACKEND}_models"
+    args = ["--model-repository", model_repository, "--tokenizer", TEST_TOKENIZER]
```

Comment:
You can mock Triton core and pass here if the init function allows passing an Triton core object, then you don't need real Triton running to test frontend behavior.

Reply:
Will follow-up with mock tests once you're happy with this general refactor: https://github.com/triton-inference-server/server/pull/7570